generators/generator-microsoft-bot-adaptive/generators/app/index.js

```diff
@@ -228,64 +152,101 @@ module.exports = class extends Generator {
   }
 
-  _copySchemas() {
-    const botName = this.options.botName;
-    const directoryName = 'schemas';
+  _writeJsPackageJson() {
+    const { botName, integration } = this.options;
 
-    this.fs.copyTpl(
-      this.templatePath(path.join('assets', directoryName)),
-      this.destinationPath(path.join(botName, directoryName)),
-      {
-        botName,
-      }
-    );
+    const dependencies = {};
+
+    switch (integration) {
+      case integrations.webapp:
+        Object.assign(dependencies, {
+          'botbuilder-runtime-integration-restify': 'next',
+          minimist: 'latest',
+        });
+
+        break;
+
+      case integrations.functions:
+        Object.assign(dependencies, {
+          'botbuilder-runtime-integration-azure-functions': 'next',
+        });
+
+        break;
+    }
+
+    this.fs.writeJSON(this.destinationPath(botName, 'package.json'), {
+      name: botName,
+      private: true,
+      scripts: {
+        build: 'echo done',
+      },
+      dependencies: Object.assign(
+        {},
+        this.packageReferences.reduce(
+          (acc, { name, version }) => Object.assign(acc, { [name]: version }),
+          {}
+        ),
+        dependencies
+      ),
+    });
   }
 
   _writeApplicationSettings() {
-    const botName = this.options.botName;
-    const fileName = 'appsettings.json';
-
-    const filePath = path.join(
+    const {
+      applicationSettingsDirectory = '.',
       botName,
-      this.applicationSettingsDirectory
-        ? this.applicationSettingsDirectory
-        : '',
-      fileName
-    );
+      integration,
+      platform,
+    } = this.options;
 
     const appSettings = this.fs.readJSON(
-      this.templatePath(path.join('assets', fileName))
+      this.templatePath('assets', 'appsettings.json')
     );
 
     appSettings.luis.name = botName;
-    appSettings.runtime.command = `dotnet run --project ${botName}.csproj`;
-    appSettings.runtime.key = 'adaptive-runtime-dotnet-webapp';
 
-    for (const pluginDefinition of this.pluginDefinitions) {
-      appSettings.runtimeSettings.plugins.push(pluginDefinition);
+    appSettings.runtime.key = `adaptive-runtime-${platform}-${integration}`;
+    appSettings.runtime.command = {
+      [platforms.dotnet]: `dotnet run --project ${botName}.csproj`,
+      [platforms.js]: 'node index.js',
+    }[platform];
+
+    for (const { name } of this.packageReferences) {
+      appSettings.runtimeSettings.plugins.push({
+        name,
+        settingsPrefix: name,
+      });
     }
 
-    if (this.modifyApplicationSettings) {
-      this.modifyApplicationSettings(appSettings);
+    for (const { name, settingsPrefix } of this.pluginDefinitions) {
+      appSettings.runtimeSettings.plugins.push({
+        name,
+        settingsPrefix: settingsPrefix || name,
+      });
     }
 
-    this.fs.writeJSON(filePath, appSettings);
+    this.fs.writeJSON(
+      this.destinationPath(
+        botName,
+        applicationSettingsDirectory,
```

Comment:
Not sure if we need to special case applicationSettingsDirectory being null or whatnot - I assume the above is assigning default value of '.'?

Reply:
Yep - we mark the type as `rt.String` and then transform the object into a partial which also allows `undefined`, but not `null`. The restructured assignment says "if `applicationSettingsDirectory` is `undefined`, set it `'.'`" which should work. So the value will never be `null` as `runtypes` will throw.

```
~/s/g/m/botframework-components/g/generator-microsoft-bot-adaptive on jpg/js-generators > node
Welcome to Node.js v12.21.0.
Type ".help" for more information.
> const rt = require('runtypes')
undefined
> rt.Record({foo: rt.String}).asPartial().check({foo: null})
Uncaught ValidationError: Expected string, but was null in foo
    at new ValidationError (/Users/jpg/src/github/microsoft/botframework-components/node_modules/runtypes/lib/errors.js:22:28)
    at Object.check (/Users/jpg/src/github/microsoft/botframework-components/node_modules/runtypes/lib/runtype.js:30:15) {
  key: 'foo'
}
> rt.Record({foo: rt.String}).asPartial().check({foo: undefined})
{ foo: undefined }
```

Reply:
Here's the full example:
```
> const {foo = 'bar'} = rt.Record({foo: rt.String}).asPartial().check({foo: undefined})
undefined
> foo
'bar'
```

Reply:
Awesome, sounds good to me!